joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/CallTests.scala

```diff
@@ -4,17 +4,39 @@ import io.joern.php2cpg.astcreation.AstCreator.NameConstants
 import io.joern.php2cpg.testfixtures.PhpCode2CpgFixture
 import io.joern.x2cpg.Defines
 import io.shiftleft.codepropertygraph.generated.{DispatchTypes, Operators}
-import io.shiftleft.codepropertygraph.generated.nodes.{Call, Identifier}
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, FieldIdentifier, Identifier, Literal}
 import io.shiftleft.semanticcpg.language.*
 
 class CallTests extends PhpCode2CpgFixture {
+
   "variable call arguments with names matching methods should not have a methodref" in {
     val cpg = code("""<?php
       |$a = file("ABC");
-      |$file = $a.contents();
+      |$file = $a->contents();
       |foo($file);
       |""".stripMargin)
 
+    val fileCall = cpg.call.nameExact("file").head
+    fileCall.code shouldBe "file(\"ABC\")"
+    fileCall.dispatchType shouldBe DispatchTypes.STATIC_DISPATCH
+    inside(fileCall.astChildren.l) { case (arg: Literal) :: Nil =>
+      arg.code shouldBe "\"ABC\""
+      arg.argumentIndex shouldBe 1
+    }
+
+    val contentsCall = cpg.call.nameExact("contents").head
+    contentsCall.code shouldBe "$a->contents()"
+    contentsCall.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+    inside(contentsCall.astChildren.l) { case (base: Call) :: (receiver: Identifier) :: Nil =>
+      base.name shouldBe Operators.fieldAccess
+      base.code shouldBe "$a->contents"
+      base.argumentIndex shouldBe -1
```

Comment:
There must have been a misunderstanding in our last talk.
The base is the "this" argument and this is just an IDENTIFIER `a` in this case.
You can reference the same IDENTIFIER node as receiver. 
I just red through the PHP docu and played around with it a little bit and i think we can model the PHP classes and the call dispatch in a classic vtable style.